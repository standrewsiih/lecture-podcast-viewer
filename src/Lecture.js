import React from 'react';
import moment from 'moment';

const Lecture = (props) => {
  return(
    <div className="lecture">
      <div className="content-panel">    
        <h2 className="page-subheading">{props.title}</h2>
        <p className="date">{moment(props.pubDate).format('dddd MMMM Do YYYY')}</p>
        <p>{props.subtitle}</p>
        <p><audio preload="none" controls="controls"><source src={props.url} type="audio/mpeg" /></audio></p>
      </div>
    </div>
  );
};

export default Lecture;