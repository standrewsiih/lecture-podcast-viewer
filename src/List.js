import React, { Component } from 'react';
import Lecture from './Lecture';

class List extends Component {
  constructor() {
    super();

    this.state = {
      sort: {
        pubDate: "asc",
        title: "asc",
        length: "asc"
      },
      filter: "all",
      search: undefined      
    };
  }

  render() {
    return (
      <div className="List">
        <p>Sort lectures by:&nbsp;
          <button id="pubDate" onClick={this.sort}>Date</button>
          <button id="title" onClick={this.sort}>Title</button>
          <button id="length" onClick={this.sort}>Duration</button>
        </p>
        <p>Show lectures from:&nbsp;
          <button id="all" onClick={this.filter}>all years</button>
          {this.filterButtons()}
        </p>
        <p>Search lectures:&nbsp;
          <input id="searchString" type="search" placeholder="Enter speaker or keyword" onChange={this.search}/>
        </p>
        {this.show()}
      </div>
    )
  };

  show = () => {
    let lecturesToShow = this.props.lectures
      .filter((lecture) => { // filter lectures
        if(this.state.filter !== "all") {
          return lecture.year === this.state.filter;
        }
        return true;
      })
      .filter((lecture) => { // search lectures
        return this.searchLecture(lecture);
      })
      .map((lecture, index) => {
        return (
          <Lecture {...lecture} key={index}/>
        )
      });

    if(lecturesToShow.length > 0) {
      return lecturesToShow;      
    }
    else {
      return (<p>No lecture descriptions matched your search terms.</p>);
    }
  };

  filterButtons = () => {
    let years = [];

    this.props.lectures.filter((lecture) => {
      if(years.indexOf(lecture.year) === -1) {
        return years.push(lecture.year);
      }
      return false;
    });        

    return years.sort().map((year, index) => {
      return (
        <button key={index} id={'y' + year} onClick={this.filter}>{year}</button>
      )
    })
  };

  sort = (e) => {    
    let type = e.target.id;

    this.props.lectures.sort((a, b) => {
      if(this.state.sort[type] === "asc") {
        if(a[type] < b[type]) return -1;
        if(a[type] > b[type]) return 1;
        return 0;
      } else {
        if(a[type] > b[type]) return -1;
        if(a[type] < b[type]) return 1;
        return 0;
      }
    });

    let sort = {...this.state.sort};
    sort[type] = (sort[type] === "asc") ? "desc" : "asc";
    
    this.setState({
      sort: sort
    });
  };

  filter = (e) => {
    let filter = e.target.id === 'all' ? e.target.id : e.target.id.substring(1);
    this.setState({
      filter: filter
    });
  };

  search = (e) => {
    let searchTerm = e.target.value.toLowerCase();

    if(searchTerm.length < 3) {
      searchTerm = undefined;
    }

    this.setState({
      search: searchTerm
    });
  };

  searchLecture = (lecture) => {
    if(this.state.search) {
      if(lecture.title.toLowerCase().includes(this.state.search)) return true;
      if(lecture.subtitle.toLowerCase().includes(this.state.search)) return true;
      return false;
    }

    return true;
  };
}
  
export default List;