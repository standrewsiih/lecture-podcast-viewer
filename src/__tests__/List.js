import React from 'react';
import { shallow } from 'enzyme';
import List from '../List';
import Lecture from '../LectureModel';

describe('List', () => {

  let props = {
    lectures: [ 
      new Lecture (
        'some title',
        'some subtitle',
        'some summary',
        new Date('Tue, 18 Apr 2017 17:15:00 GMT'),
        '56:54',
        'https://www.intellectualhistory.net/some_lecture.mp3'),
      new Lecture (
        'another title',
        'another subtitle',
        'another summary',
        new Date('Tue, 1 Nov 2016 17:15:00 GMT'),
        '50:22',
        'https://www.intellectualhistory.net/another_lecture.mp3'),
      new Lecture (
        'some other title',
        'some other subtitle',
        'some other summary',
        new Date('Tue, 23 Jun 2015 17:15:00 GMT'),
        '52:14',
        'https://www.intellectualhistory.net/some_other_file.mp3'),
      new Lecture (
        'some other other title',
        'some other other subtitle',
        'some other other summary',
        new Date('Tue, 30 Jun 2015 17:15:00 GMT'),
        '55:52')
      ]
  };

  let list = shallow(<List {...props}/>);
  
  it('should render properly', () => {      
    expect(list).toMatchSnapshot();
  });

  it('should display Lecture components', () => {
    expect(list.find('Lecture')).toHaveLength(4);
  });

  describe('sort function', () => {

    it('should sort the Lecture components by title descending', () => {
      list.find('button').find('#title').simulate('click', { target: { id: 'title' }});

      expect(list.state().sort.title).toBe('desc');      
      expect(list.find('Lecture').first().props().title).toEqual('another title');
    });

    it('should sort the Lecture components by title ascending', () => {
      list.find('button').find('#title').simulate('click', { target: { id: 'title' }});

      expect(list.state().sort.title).toBe('asc');
      expect(list.find('Lecture').first().props().title).toEqual('some title');
    });
    
    it('should sort the Lecture components by date descending', () => {
      list.find('button').find('#pubDate').simulate('click', { target: { id: 'pubDate' }});

      expect(list.state().sort.pubDate).toBe('desc');
      expect(list.find('Lecture').first().props().pubDate.toString()).toEqual('Tue Jun 23 2015 18:15:00 GMT+0100 (BST)');
    });

    it('should sort the Lecture components by date ascending', () => {
      list.find('button').find('#pubDate').simulate('click', { target: { id: 'pubDate' }});

      expect(list.state().sort.pubDate).toBe('asc');      
      expect(list.find('Lecture').first().props().pubDate.toString()).toEqual('Tue Apr 18 2017 18:15:00 GMT+0100 (BST)');
    });
    
  });

  describe('search function', () => {

    it('should search the lecture title for the specified string', () => {
      list.find('input').find('#searchString').simulate('change', { target: { value: 'some other title' }});

      expect(list.find('Lecture')).toHaveLength(1);
      expect(list.find('Lecture').first().props().title).toContain('some other title');
    });

    it('should search the lecture subtitle for the specified string', () => {
      list.find('input').find('#searchString').simulate('change', { target: { value: 'some subtitle' }});

      expect(list.find('Lecture')).toHaveLength(1);
      expect(list.find('Lecture').first().props().subtitle).toContain('some subtitle');
    });

    it('should convert the search string to lower case', () => {
      list.find('input').find('#searchString').simulate('change', { target: { value: 'SOME OTHER TITLE' }});

      expect(list.state().search).toEqual('some other title');
    });

    it('should display no lectures when there are no search results', () => {
      list.find('input').find('#searchString').simulate('change', { target: { value: 'szxcxzcxc' }});

      expect(list.find('Lecture')).toHaveLength(0);
    });

    it('should not search if the string is less than 3 characters', () => {
      list.find('input').find('#searchString').simulate('change', { target: { value: 'so' }});

      expect(list.state().search).toBe(undefined);
    });
  });

  describe('filter function', () => {
    it('should display lectures from a specific year', () => {        
      list.find('button').find('#y2015').simulate('click', { target: { id: 'y2015' }});

      expect(list.find('Lecture')).toHaveLength(2);
    });

    it('should display all lectures', () => {        
      list.find('button').find('#all').simulate('click', { target: { id: 'all' }});

      expect(list.find('Lecture')).toHaveLength(4);
    });
  });    

});