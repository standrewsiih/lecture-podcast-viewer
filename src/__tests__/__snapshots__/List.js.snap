// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    lectures={
        Array [
            Lecture {
              "calcLength": [Function],
              "duration": "56:54",
              "formatSubtitle": [Function],
              "length": 3414,
              "pubDate": 2017-04-18T17:15:00.000Z,
              "subtitle": "some subtitle.",
              "summary": "some summary",
              "title": "some title",
              "url": "https://www.intellectualhistory.net/some_lecture.mp3",
              "year": "2017",
            },
            Lecture {
              "calcLength": [Function],
              "duration": "50:22",
              "formatSubtitle": [Function],
              "length": 3022,
              "pubDate": 2016-11-01T17:15:00.000Z,
              "subtitle": "another subtitle.",
              "summary": "another summary",
              "title": "another title",
              "url": "https://www.intellectualhistory.net/another_lecture.mp3",
              "year": "2016",
            },
            Lecture {
              "calcLength": [Function],
              "duration": "52:14",
              "formatSubtitle": [Function],
              "length": 3134,
              "pubDate": 2015-06-23T17:15:00.000Z,
              "subtitle": "some other subtitle.",
              "summary": "some other summary",
              "title": "some other title",
              "url": "https://www.intellectualhistory.net/some_other_file.mp3",
              "year": "2015",
            },
            Lecture {
              "calcLength": [Function],
              "duration": "55:52",
              "formatSubtitle": [Function],
              "length": 3352,
              "pubDate": 2015-06-30T17:15:00.000Z,
              "subtitle": "some other other subtitle.",
              "summary": "some other other summary",
              "title": "some other other title",
              "url": undefined,
              "year": "2015",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Sort lectures by: 
          <button
                    id="pubDate"
                    onClick={[Function]}
          >
                    Date
          </button>
          <button
                    id="title"
                    onClick={[Function]}
          >
                    Title
          </button>
          <button
                    id="length"
                    onClick={[Function]}
          >
                    Duration
          </button>
</p>,
        <p>
          Show lectures from: 
          <button
                    id="all"
                    onClick={[Function]}
          >
                    all years
          </button>
          <button
                    id="y2015"
                    onClick={[Function]}
          >
                    2015
          </button>
          <button
                    id="y2016"
                    onClick={[Function]}
          >
                    2016
          </button>
          <button
                    id="y2017"
                    onClick={[Function]}
          >
                    2017
          </button>
</p>,
        <p>
          Search lectures: 
          <input
                    id="searchString"
                    onChange={[Function]}
                    placeholder="Enter speaker or keyword"
                    type="search"
          />
</p>,
        Array [
          <Lecture
            calcLength={[Function]}
            duration="56:54"
            formatSubtitle={[Function]}
            length={3414}
            pubDate={2017-04-18T17:15:00.000Z}
            subtitle="some subtitle."
            summary="some summary"
            title="some title"
            url="https://www.intellectualhistory.net/some_lecture.mp3"
            year="2017"
/>,
          <Lecture
            calcLength={[Function]}
            duration="50:22"
            formatSubtitle={[Function]}
            length={3022}
            pubDate={2016-11-01T17:15:00.000Z}
            subtitle="another subtitle."
            summary="another summary"
            title="another title"
            url="https://www.intellectualhistory.net/another_lecture.mp3"
            year="2016"
/>,
          <Lecture
            calcLength={[Function]}
            duration="52:14"
            formatSubtitle={[Function]}
            length={3134}
            pubDate={2015-06-23T17:15:00.000Z}
            subtitle="some other subtitle."
            summary="some other summary"
            title="some other title"
            url="https://www.intellectualhistory.net/some_other_file.mp3"
            year="2015"
/>,
          <Lecture
            calcLength={[Function]}
            duration="55:52"
            formatSubtitle={[Function]}
            length={3352}
            pubDate={2015-06-30T17:15:00.000Z}
            subtitle="some other other subtitle."
            summary="some other other summary"
            title="some other other title"
            url={undefined}
            year="2015"
/>,
        ],
      ],
      "className": "List",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sort lectures by: ",
            <button
              id="pubDate"
              onClick={[Function]}
>
              Date
</button>,
            <button
              id="title"
              onClick={[Function]}
>
              Title
</button>,
            <button
              id="length"
              onClick={[Function]}
>
              Duration
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sort lectures by: ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Date",
              "id": "pubDate",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Date",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Title",
              "id": "title",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Title",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Duration",
              "id": "length",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Duration",
            "type": "button",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Show lectures from: ",
            <button
              id="all"
              onClick={[Function]}
>
              all years
</button>,
            Array [
              <button
                id="y2015"
                onClick={[Function]}
>
                2015
</button>,
              <button
                id="y2016"
                onClick={[Function]}
>
                2016
</button>,
              <button
                id="y2017"
                onClick={[Function]}
>
                2017
</button>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Show lectures from: ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "all years",
              "id": "all",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "all years",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "2015",
              "id": "y2015",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "2015",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "2016",
              "id": "y2016",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "2016",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "2017",
              "id": "y2017",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "2017",
            "type": "button",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Search lectures: ",
            <input
              id="searchString"
              onChange={[Function]}
              placeholder="Enter speaker or keyword"
              type="search"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Search lectures: ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "id": "searchString",
              "onChange": [Function],
              "placeholder": "Enter speaker or keyword",
              "type": "search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "calcLength": [Function],
          "duration": "56:54",
          "formatSubtitle": [Function],
          "length": 3414,
          "pubDate": 2017-04-18T17:15:00.000Z,
          "subtitle": "some subtitle.",
          "summary": "some summary",
          "title": "some title",
          "url": "https://www.intellectualhistory.net/some_lecture.mp3",
          "year": "2017",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "calcLength": [Function],
          "duration": "50:22",
          "formatSubtitle": [Function],
          "length": 3022,
          "pubDate": 2016-11-01T17:15:00.000Z,
          "subtitle": "another subtitle.",
          "summary": "another summary",
          "title": "another title",
          "url": "https://www.intellectualhistory.net/another_lecture.mp3",
          "year": "2016",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "calcLength": [Function],
          "duration": "52:14",
          "formatSubtitle": [Function],
          "length": 3134,
          "pubDate": 2015-06-23T17:15:00.000Z,
          "subtitle": "some other subtitle.",
          "summary": "some other summary",
          "title": "some other title",
          "url": "https://www.intellectualhistory.net/some_other_file.mp3",
          "year": "2015",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "calcLength": [Function],
          "duration": "55:52",
          "formatSubtitle": [Function],
          "length": 3352,
          "pubDate": 2015-06-30T17:15:00.000Z,
          "subtitle": "some other other subtitle.",
          "summary": "some other other summary",
          "title": "some other other title",
          "url": undefined,
          "year": "2015",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Sort lectures by: 
            <button
                        id="pubDate"
                        onClick={[Function]}
            >
                        Date
            </button>
            <button
                        id="title"
                        onClick={[Function]}
            >
                        Title
            </button>
            <button
                        id="length"
                        onClick={[Function]}
            >
                        Duration
            </button>
</p>,
          <p>
            Show lectures from: 
            <button
                        id="all"
                        onClick={[Function]}
            >
                        all years
            </button>
            <button
                        id="y2015"
                        onClick={[Function]}
            >
                        2015
            </button>
            <button
                        id="y2016"
                        onClick={[Function]}
            >
                        2016
            </button>
            <button
                        id="y2017"
                        onClick={[Function]}
            >
                        2017
            </button>
</p>,
          <p>
            Search lectures: 
            <input
                        id="searchString"
                        onChange={[Function]}
                        placeholder="Enter speaker or keyword"
                        type="search"
            />
</p>,
          Array [
            <Lecture
              calcLength={[Function]}
              duration="56:54"
              formatSubtitle={[Function]}
              length={3414}
              pubDate={2017-04-18T17:15:00.000Z}
              subtitle="some subtitle."
              summary="some summary"
              title="some title"
              url="https://www.intellectualhistory.net/some_lecture.mp3"
              year="2017"
/>,
            <Lecture
              calcLength={[Function]}
              duration="50:22"
              formatSubtitle={[Function]}
              length={3022}
              pubDate={2016-11-01T17:15:00.000Z}
              subtitle="another subtitle."
              summary="another summary"
              title="another title"
              url="https://www.intellectualhistory.net/another_lecture.mp3"
              year="2016"
/>,
            <Lecture
              calcLength={[Function]}
              duration="52:14"
              formatSubtitle={[Function]}
              length={3134}
              pubDate={2015-06-23T17:15:00.000Z}
              subtitle="some other subtitle."
              summary="some other summary"
              title="some other title"
              url="https://www.intellectualhistory.net/some_other_file.mp3"
              year="2015"
/>,
            <Lecture
              calcLength={[Function]}
              duration="55:52"
              formatSubtitle={[Function]}
              length={3352}
              pubDate={2015-06-30T17:15:00.000Z}
              subtitle="some other other subtitle."
              summary="some other other summary"
              title="some other other title"
              url={undefined}
              year="2015"
/>,
          ],
        ],
        "className": "List",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sort lectures by: ",
              <button
                id="pubDate"
                onClick={[Function]}
>
                Date
</button>,
              <button
                id="title"
                onClick={[Function]}
>
                Title
</button>,
              <button
                id="length"
                onClick={[Function]}
>
                Duration
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sort lectures by: ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Date",
                "id": "pubDate",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Date",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Title",
                "id": "title",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Title",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Duration",
                "id": "length",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Duration",
              "type": "button",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Show lectures from: ",
              <button
                id="all"
                onClick={[Function]}
>
                all years
</button>,
              Array [
                <button
                  id="y2015"
                  onClick={[Function]}
>
                  2015
</button>,
                <button
                  id="y2016"
                  onClick={[Function]}
>
                  2016
</button>,
                <button
                  id="y2017"
                  onClick={[Function]}
>
                  2017
</button>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Show lectures from: ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "all years",
                "id": "all",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "all years",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "2015",
                "id": "y2015",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "2015",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "2016",
                "id": "y2016",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "2016",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "2017",
                "id": "y2017",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "2017",
              "type": "button",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Search lectures: ",
              <input
                id="searchString"
                onChange={[Function]}
                placeholder="Enter speaker or keyword"
                type="search"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Search lectures: ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "id": "searchString",
                "onChange": [Function],
                "placeholder": "Enter speaker or keyword",
                "type": "search",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "calcLength": [Function],
            "duration": "56:54",
            "formatSubtitle": [Function],
            "length": 3414,
            "pubDate": 2017-04-18T17:15:00.000Z,
            "subtitle": "some subtitle.",
            "summary": "some summary",
            "title": "some title",
            "url": "https://www.intellectualhistory.net/some_lecture.mp3",
            "year": "2017",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "calcLength": [Function],
            "duration": "50:22",
            "formatSubtitle": [Function],
            "length": 3022,
            "pubDate": 2016-11-01T17:15:00.000Z,
            "subtitle": "another subtitle.",
            "summary": "another summary",
            "title": "another title",
            "url": "https://www.intellectualhistory.net/another_lecture.mp3",
            "year": "2016",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "calcLength": [Function],
            "duration": "52:14",
            "formatSubtitle": [Function],
            "length": 3134,
            "pubDate": 2015-06-23T17:15:00.000Z,
            "subtitle": "some other subtitle.",
            "summary": "some other summary",
            "title": "some other title",
            "url": "https://www.intellectualhistory.net/some_other_file.mp3",
            "year": "2015",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "calcLength": [Function],
            "duration": "55:52",
            "formatSubtitle": [Function],
            "length": 3352,
            "pubDate": 2015-06-30T17:15:00.000Z,
            "subtitle": "some other other subtitle.",
            "summary": "some other other summary",
            "title": "some other other title",
            "url": undefined,
            "year": "2015",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
